src/smc-webapp/editor.coffee

```diff
@@ -1624,6 +1624,12 @@ class CodeMirrorEditor extends FileEditor
                 evaluate_key = "Shift-Enter"
             extraKeys[evaluate_key] = (editor)   => @action_key(execute: true, advance:true, split:false)
 
+        # turn off autoCloseBrackets for haskell files SMC #972
+        if misc.filename_extension_notilde(filename) == "hs" || misc.filename_extension_notilde(filename) == "lhs"
```

Comment:
It's a little ugly to call `misc.filename_extension_notilde(filename)` twice (also `or` is same as `||` in coffeescript).  Does `misc.filename_extension_notilde(filename) in ['hs', 'lhs']` work?  Could you just make the line `autoCloseBrackets       : opts.auto_close_brackets` be 
```
autoCloseBrackets       : opts.auto_close_brackets and (misc.filename_extension_notilde(filename) not in ['hs', 'lhs'])
```
?